hypervisor/common/hypercall.c

```diff
@@ -150,11 +150,13 @@ int32_t hcall_create_vm(struct acrn_vm *vm, uint64_t param)
 	struct acrn_vm *target_vm = NULL;
 	struct acrn_create_vm cv;
 	struct acrn_vm_config* vm_config = NULL;
+	bool valid;
 
 	(void)memset((void *)&cv, 0U, sizeof(cv));
 	if (copy_from_gpa(vm, &cv, param, sizeof(cv)) == 0) {
 		vm_id = get_vmid_by_uuid(&cv.uuid[0]);
-		if (vm_id < CONFIG_MAX_VM_NUM) {
+		valid = is_valid_vm(get_vm_from_vmid(vm_id));
+		if (vm_id < CONFIG_MAX_VM_NUM && !valid) {
```

Comment:
BTW, can you add a @pre at create_vm to make sure vm->state == VM_STATE_INVALID？

Reply:
OK